phoenix-core/src/main/java/org/apache/phoenix/iterate/ScanningResultIterator.java

```diff
@@ -173,6 +177,12 @@ public Tuple next() throws SQLException {
         try {
             Result result = scanner.next();
             while (result != null && (result.isEmpty() || isDummy(result))) {
+                long timeOutForScan = maxQueryEndTime - EnvironmentEdgeManager.currentTimeMillis();
+                if (timeOutForScan < 0) {
+                    throw new SQLExceptionInfo.Builder(OPERATION_TIMED_OUT).setMessage(
+                            ". Query couldn't be completed in the allotted time at ScanningResultIterator: "
```

Comment:
nit: We probably shouldn't be mentioning Phoenix internals like ScanningResultIterator in an error message that goes back to end-users.